engine/hatchery/swarm/swarm.go

```diff
@@ -404,6 +441,65 @@ func (h *HatcherySwarm) createAndStartContainer(name, image, network, networkAli
 	return nil
 }
 
+var regexPort = regexp.MustCompile("^--port=(.*):(.*)$")
+
+type dockerOpts struct {
+	ports      map[docker.Port][]docker.PortBinding
+	priviledge bool
+}
+
+func computeDockerOpts(isSharedInfra bool, requirements []sdk.Requirement) (*dockerOpts, error) {
+	dockerOpts := &dockerOpts{}
+
+	for _, r := range requirements {
+		if r.Type == sdk.ModelRequirement {
+			// args are separated by a space
+			// example: golang:1.9.1 --port=8080:8080/tcp
+			for idx, opt := range strings.Split(r.Value, " ") {
+				if idx == 0 {
+					continue // it's image name
+				}
+				if isSharedInfra {
+					return nil, fmt.Errorf("You could not use this docker options '%s' with a 'shared.infra' hatchery. Please use you own hatchery or remove this option.", opt)
+				}
+				if strings.HasPrefix(opt, "--port=") {
+					if err := dockerOpts.computeDockerOptsPorts(opt); err != nil {
+						return nil, err
+					}
+				} else if opt == "--priviledge" {
+					dockerOpts.priviledge = true
+				} else {
+					return nil, fmt.Errorf("Options not supported: %s", opt)
+				}
+			}
+		}
+	}
+
+	return dockerOpts, nil
+}
+
+func (d *dockerOpts) computeDockerOptsPorts(arg string) error {
+	if regexPort.MatchString(arg) {
+		s := regexPort.FindStringSubmatch(arg)
+		//s = --port=8081:8182/tcp // hostPort:containerPort
+		//s[0] = --port=8081:8182/tcp
+		//s[1] = 8081 // hostPort
+		//s[2] = 8182/tcp  // containerPort
+		containerPort := s[2]
```

Comment:
is could be nice to be tcp per default (id s[2] has not suffix /.*, add /tcp)

Reply:
done

Reply:
here: https://github.com/ovh/cds/pull/1481/commits/e8a30ef6304099bac78a8d7bdfafd25e10945b57#diff-5b928cb03058ec28cb0709fcac0a56b6R497